app/models/miq_queue.rb

```diff
@@ -62,7 +62,6 @@ def self.messaging_client_options
     return if opts.nil?
 
     opts.transform_values! { |v| v.kind_of?(String) ? ManageIQ::Password.try_decrypt(v) : v }
-    opts.merge(:encoding => "json", :protocol => messaging_protocol)
```

Comment:
Unclear why we were merging these into the options loaded from the yaml files rather than simply storing them in the yaml files.

`:protocol` specifically made no sense because you could run into a situation where you had copied the `config/messaging.kafka.yml` file to `config/messaging.yml` but then set `messaging_type` to `artemis`.  Better to just let the `messaging.yml` drive all of the options.

Reply:
I suspect this was actually for the podified case where we don't have the whole messaging.yml file, but we have a separate method for loading options from env vars so I moved these "default" options there